pkg/providers/aws/provider_redis.go

```diff
@@ -0,0 +1,282 @@
+package aws
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/aws/aws-sdk-go/service/elasticache/elasticacheiface"
+	"github.com/openshift/origin/Godeps/_workspace/src/github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/sirupsen/logrus"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/elasticache"
+	"github.com/integr8ly/cloud-resource-operator/pkg/apis/integreatly/v1alpha1"
+	"github.com/integr8ly/cloud-resource-operator/pkg/resources"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/integr8ly/cloud-resource-operator/pkg/providers"
+
+	errorUtil "github.com/pkg/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	defaultCacheNodeType     = "cache.t2.micro"
+	defaultEngineVersion     = "2.8.24"
```

Comment:
May need to change these values, believe that 3scale uses redis version 3.2.13, but AWS only supports 3.2.10 see https://docs.aws.amazon.com/AmazonElastiCache/latest/red-ug/supported-engine-versions.html for more info